cylc/flow/wallclock.py

```diff
@@ -85,9 +87,9 @@ def now(override_use_utc=None):
 
     """
     if override_use_utc or (override_use_utc is None and _FLAGS['utc_mode']):
-        return datetime.utcnow(), False
+        return datetime.now(timezone.utc), False
```

Comment:
There is a difference between the old and new methods:

```python
>>> str(datetime.utcnow())
'2025-02-28 13:18:42.595223'
>>> str(datetime.now(timezone.utc))
'2025-02-28 13:18:54.379479+00:00'
```

We may be stringifying datetimes like this, e.g. for the database. Need to make sure it doesn't cause issues.

Reply:
I've checked all uses of `cylc.flow.wallclock.now()` and we're not directly stringifying the native `datetime` object it returns anywhere. I've also added a test for `get_time_string()` to ensure it doesn't treat naive and tz-aware `datetime`s any differently. (Actually a lot of the logic in the wallclock module could be simplified to make better use of `datetime`'s handling of time zones but that will be for another day)

Reply:
> we're not directly stringifying 

:+1:

> could be simplified

This might be true, but be aware we support alternative calendars that datetime does not which might explain this?